packages/firestore/test/unit/specs/listen_spec.test.ts

```diff
@@ -1236,4 +1236,164 @@ describeSpec('Listens:', [], () => {
+    }
+  );
+
+  specTest(
+    'onSnapshotsInSync fires when called even if there are no local listeners',
+    [],
+    () => {
+      return spec()
+        .userAddsSnapshotsInSyncListener()
+        .expectSnapshotsInSyncEvent()
+        .userAddsSnapshotsInSyncListener()
+        .expectSnapshotsInSyncEvent();
+    }
+  );
+
+  specTest('onSnapshotsInSync fires for metadata changes', [], () => {
+    const query = Query.atPath(path('collection'));
+    const docAv1 = doc('collection/a', 1000, { v: 1 });
+    const docAv2Local = doc(
+      'collection/a',
+      1000,
+      { v: 2 },
+      { hasLocalMutations: true }
+    );
+    const docAv2 = doc('collection/a', 2000, { v: 2 });
+
+    return spec()
+      .userListens(query)
+      .watchAcksFull(query, 1000, docAv1)
+      .expectEvents(query, { added: [docAv1] })
+      .userAddsSnapshotsInSyncListener()
+      .expectSnapshotsInSyncEvent()
+      .userSets('collection/a', { v: 2 })
+      .expectEvents(query, {
+        hasPendingWrites: true,
+        modified: [docAv2Local]
+      })
+      .expectSnapshotsInSyncEvent()
+      .watchSends({ affects: [query] }, docAv2)
+      .watchSnapshots(2000)
+      .writeAcks('collection/a', 2000)
+      .expectEvents(query, {
+        metadata: [docAv2]
+      })
+      .expectSnapshotsInSyncEvent();
+  });
+
+  specTest(
+    'onSnapshotsInSync fires once for multiple event snapshots',
+    [],
+    () => {
+      const query1 = Query.atPath(path('collection'));
+      const query2 = Query.atPath(path('collection/a'));
+      const docAv1 = doc('collection/a', 1000, { v: 1 });
+      const docAv2Local = doc(
+        'collection/a',
+        1000,
+        { v: 2 },
+        { hasLocalMutations: true }
+      );
+      const docAv2 = doc('collection/a', 2000, { v: 2 });
+
+      return spec()
+        .userListens(query1)
+        .watchAcksFull(query1, 1000, docAv1)
+        .expectEvents(query1, { added: [docAv1] })
+        .userListens(query2)
+        .expectEvents(query2, { fromCache: true, added: [docAv1] })
+        .watchAcksFull(query2, 1000, docAv1)
+        .expectEvents(query2, { fromCache: false })
+        .userAddsSnapshotsInSyncListener()
+        .expectSnapshotsInSyncEvent()
+        .userSets('collection/a', { v: 2 })
+        .expectEvents(query1, {
+          hasPendingWrites: true,
+          modified: [docAv2Local]
+        })
+        .expectEvents(query2, {
+          hasPendingWrites: true,
+          modified: [docAv2Local]
+        })
+        .expectSnapshotsInSyncEvent()
+        .watchSends({ affects: [query1, query2] }, docAv2)
+        .watchSnapshots(2000)
+        .writeAcks('collection/a', 2000)
+        .expectEvents(query1, {
+          metadata: [docAv2]
+        })
+        .expectEvents(query2, {
+          metadata: [docAv2]
+        })
+        .expectSnapshotsInSyncEvent();
+    }
+  );
+
+  specTest(
+    'onSnapshotsInSync fires for multiple listeners',
+    ['exclusive'],
```

Comment:
This mean the rest of the tests aren't being run. :-)

Reply:
ahhhhhh sorry for all the careless errors I've been leaving around in this PR 😢